src/mesh/mesh_communication.C

```diff
@@ -1001,9 +1001,6 @@ void MeshCommunication::gather (const processor_id_type, DistributedMesh &) cons
 // ------------------------------------------------------------
 void MeshCommunication::gather (const processor_id_type root_id, DistributedMesh & mesh) const
 {
-  // The mesh should know it's about to be serialized
-  libmesh_assert (mesh.is_serial());
```

Comment:
Ok  - I'll do it.  I kind of just thought it was overzealous.  Even in the case where you really are doing a full serialization you may not want to actually tell the mesh it's been serialized...